src/Runner.Worker/ActionManager.cs

```diff
@@ -986,16 +1149,16 @@ private void ConfigureAuthorizationFromContext(IExecutionContext executionContex
 
         private string GetWatermarkFilePath(string directory) => directory + ".completed";
 
-        private ActionContainer PrepareRepositoryActionAsync(IExecutionContext executionContext, Pipelines.ActionStep repositoryAction)
+        private ActionSetupInfo PrepareRepositoryActionAsync(IExecutionContext executionContext, Pipelines.ActionStep repositoryAction)
         {
             var repositoryReference = repositoryAction.Reference as Pipelines.RepositoryPathReference;
             if (string.Equals(repositoryReference.RepositoryType, Pipelines.PipelineConstants.SelfAlias, StringComparison.OrdinalIgnoreCase))
             {
                 Trace.Info($"Repository action is in 'self' repository.");
                 return null;
             }
-
-            var setupInfo = new ActionContainer();
+            var setupInfo = new ActionSetupInfo();
+            var actionContainer = new ActionContainer();
```

Comment:
should this move into each if block

Reply:
Doesn't actually make a lot of sense to do here because we use it right away

Reply:
did this change? it looks like it's not used right away

Reply:
specifically the local variable `actionContainer`

asking, because i found myself scrolling back and forth to try and understand this local variable. Looks like it's only used further below within each if block.

Reply:
Its used in this block https://github.com/actions/runner/pull/1170/files/704b39965b40c8b04be58576fbafa66e895b16e4#diff-5b282ef924f268139b4015eb491825d3e35bfa6cbe46962e3bef3bd8fa05dfccR1170 and the else side of that block. Which sets the variable at this level, then as you go into the if conditions its used more. We could refactor those linked blocks to be in every block that has `actionContainer`, but i'm not sure that makes a lot of sense for this use case, I think this is the right level to initialize this variable.

Reply:
oh :(